crates/router/src/core/routing.rs

```diff
@@ -909,26 +909,29 @@ pub async fn retrieve_default_fallback_algorithm_for_profile(
 }
 
 #[cfg(feature = "v1")]
-
 pub async fn retrieve_default_routing_config(
     state: SessionState,
-    merchant_account: domain::MerchantAccount,
+    profile_id: Option<common_utils::id_type::ProfileId>,
     transaction_type: &enums::TransactionType,
 ) -> RouterResponse<Vec<routing_types::RoutableConnectorChoice>> {
+    let profile_id = profile_id.ok_or(errors::ApiErrorResponse::GenericNotFoundError {
+        message: "profile_id not found".to_string(),
+    })?;
```

Comment:
Can we also handle the case where `profile_id` is not present.

Is can be if...else, and if `profile_id` is not present it will work like before.